### PR TITLE
DIABLO-26, bulk create/update for Salesforce contacts and courses

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/DIABLO-26

Using bulk operations (see https://github.com/simple-salesforce/simple-salesforce) gets us fewer lines of code. I've also removed the generate-csv-report per CDM run.  If you want CSV report then run the new `salesforce/all_eligible_courses` report. (More pytests coming soon.)

Given simpler bulk-upsert code, I removed the clunky `skip_salesforce_update` config.

### CODE REVIEW

**Comment by johncrossman** [comment]
Diablo-CDM  now runs in less than a minute. Prior to using bulk upserts, it was 4-5 minutes.

File: diablo/externals/salesforce.py [hunk 1]
```diff
@@ -42,146 +40,58 @@
...
-        logging.info(f'Salesforce Contact.update. {existing_record_id_}: {updated_record_}')
-    else:
-        sf.Contact.update(existing_record_id_, updated_record_)
+# def get_course_instructors(salesforce_record):
+#     ids_ = []
+#     for index_ in range(6):
+#         id_ = salesforce_record[f'Instructor_{index_ + 1}__c']
+#         if id_:
+#             ids_.append(id_)
+#     return sorted(ids_)
```
Comment (paulkerschen) [inline diablo/externals/salesforce.py]: @johncrossman Is this method hanging around as a TODO of some sort?

Comment (johncrossman) [reply diablo/externals/salesforce.py]: Whoops, I meant to delete all that. Amending.

**Comment by paulkerschen** [comment]
download

**Pull request merged**